platform/base/core/src/main/java/com/peregrine/replication/Replication.java

```diff
@@ -84,8 +85,8 @@ default void prepare(final Resource source, final boolean deep)
     List<Resource> deactivate(Resource source)
         throws ReplicationException;
 
-    default List<Resource> prepare(Collection<Resource> resourceList) throws ReplicationException {
-        return Collections.emptyList();
+    default List<Resource> prepare(final Collection<Resource> resources) throws ReplicationException {
+        return new LinkedList<>(resources);
     }
```

Comment:
I got burned by this. Please get rid of the default implementation. 
List<Resource> prepare(Collection<Resource> resourceList) throws ReplicationException;